### PR TITLE
[O365] Check contentExpiration timestamp before requesting contents

### PR DESCRIPTION
**Pull request by kcreddy**
## Proposed commit message
- Check contentExpiration timestamp before requesting contents.
  - Add system tests to verify this scenario
- Remove outermost reference to `now` as the minimum `kibana.version` is already updated to `8.12`
- Avoid state's timestamp being stuck with older than 7 days by checking it against `(now - duration("167h55m"))`

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally
`elastic-package stack down && elastic-package build && elastic-package stack up -d -v && eval "$(elastic-package stack shellinit)" && elastic-package test system --generate  -v `
```
--- Test results for package: o365 - START ---
╭─────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├─────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ o365    │ audit       │ system    │ cel       │ PASS   │ 39.764056792s │
╰─────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: o365 - END   ---
Done
```

### CODE REVIEW

File: packages/o365/data_stream/audit/agent/stream/cel.yml.hbs [hunk 1]
```diff
@@ -128,42 +127,79 @@ program: |
...
-                                          )
-                                      )
-                                  })
+                            // contents exist to consume
+                            list_contents_resp_body.map(l1, 
+                                (has(l1.contentExpiration) && (l1.contentExpiration.parse_time(time_layout.RFC3339) >= now)) ?
+                                (
+                                    request("GET", l1.contentUri).do_request().as(content_resp,
+                                        (has(content_resp.StatusCode) && content_resp.StatusCode == 200) ?
+                                        (
```
Comment (efd6) [inline packages/o365/data_stream/audit/agent/stream/cel.yml.hbs:136]: Indent this block one level.

File: packages/o365/data_stream/audit/agent/stream/cel.yml.hbs [hunk 2]
```diff
@@ -128,42 +127,79 @@ program: |
...
+                                                : (has(list_contents_resp.Header) && has(list_contents_resp.Header.Nextpageuri)) ?
+                                                    list_contents_resp.Header.Nextpageuri[0]
+                                                :
+                                                    ""
+                                            )
+                                        }
+                                    )
+                                )
+                                : 
+                                {
```
Comment (efd6) [inline packages/o365/data_stream/audit/agent/stream/cel.yml.hbs:186]: Indent this block one level.

File: packages/o365/data_stream/audit/agent/stream/cel.yml.hbs [hunk 3]
```diff
@@ -128,42 +127,79 @@ program: |
...
+                                    "events_per_content_type": [],
+                                    "content_type": content_type,
+                                    "content_created_at": {"temp": list_contents_resp_body}.collate("temp.contentCreated").max().as(temp_max, temp_max.parse_time(time_layout.RFC3339) > (now - duration("167h55m")) ? temp_max : (now - duration("167h55m")).format(time_layout.RFC3339)),
+                                    "next_page": (
+                                        (has(list_contents_resp.Header) && has(list_contents_resp.Header.NextPageUri) ) ?
+                                            list_contents_resp.Header.NextPageUri[0]
+                                        : (has(list_contents_resp.Header) && has(list_contents_resp.Header.Nextpageuri)) ?
+                                            list_contents_resp.Header.Nextpageuri[0]
+                                        :
+                                            ""
```
Comment (efd6) [inline packages/o365/data_stream/audit/agent/stream/cel.yml.hbs:196]: Since we are now on 8.12.
```suggestion
                                    "next_page": list_contents_resp.?Header.NextPageUri.orValue(list_contents_resp.?Header.Nextpageuri).orValue(""),
```
(we can probably get nice cleanup throughout by using these, but lets do that in another PR, rather than in a bugfix)

File: packages/o365/data_stream/audit/agent/stream/cel.yml.hbs [hunk 4]
```diff
@@ -128,42 +127,79 @@ program: |
...
+                                                    )
+                                                )
+                                            })
+                                        )
+                                        : 
+                                        {
+                                            "events_per_content_type": [],
+                                            "content_type": content_type,
+                                            "content_created_at": {"temp": list_contents_resp_body}.collate("temp.contentCreated").max().as(temp_max, temp_max.parse_time(time_layout.RFC3339) > (now - duration("167h55m")) ? temp_max : (now - duration("167h55m")).format(time_layout.RFC3339)),
+                                            "next_page": (
```
Comment (efd6) [inline packages/o365/data_stream/audit/agent/stream/cel.yml.hbs:174]: See comment below for use of optional types.

File: packages/o365/data_stream/audit/agent/stream/cel.yml.hbs (hunk 3 above)
Comment (kcreddy) [reply packages/o365/data_stream/audit/agent/stream/cel.yml.hbs:196]: Agreed. I was thinking of the same in another PR.

File: packages/o365/data_stream/audit/agent/stream/cel.yml.hbs (hunk 4 above)
Comment (kcreddy) [reply packages/o365/data_stream/audit/agent/stream/cel.yml.hbs]: Will do the optional types conversion in another PR 👍🏼

File: packages/o365/data_stream/audit/agent/stream/cel.yml.hbs (hunk 1 above)
Comment (kcreddy) [reply packages/o365/data_stream/audit/agent/stream/cel.yml.hbs]: indented the block

**Review by efd6 — APPROVED** [review]
